app/views/CreditAndRefunds.scala.html

```diff
@@ -82,162 +65,191 @@
     }
 }
 
+@creditAndRefundRefundProgressHtml(amount: BigDecimal) = {
+    @if(messages.lang.language.toLowerCase().equals("cy")) {
+        <li>
+            <p class="govuk-heading-s">
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress-prt-1")</span>
+                @amount.toCurrencyString
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress-prt-2")</span>
+            </p>
+        </li>
+    } else {
+        <li>
+            <p class="govuk-heading-s">@amount.toCurrencyString
+                <span class="govuk-body">@messages("credit-and-refund.refundProgress")</span>
+            </p>
+        </li>
+    }
+}
+
+@maybeCreditChargesHead = @{
+    creditCharges.headOption.getOrElse(throw MissingFieldException("Credit Charges"))
+}
+
+@maybeBalancesDetails = @{
+    balance.getOrElse(throw MissingFieldException("Balance"))
+}
+
+@maybeAvailableCredit = @{
+    maybeBalancesDetails.availableCredit.getOrElse(throw MissingFieldException("Available Credit"))
+}
+
 @mainTemplate(
     pageTitle = messages("credit-and-refund.heading"),
     backUrl = Some(backUrl),
     isAgent = isAgent,
     useFallbackBackLink = true
 ) {
 
-
     @h1(msg = "credit-and-refund.heading")
 
     @if(balance.isDefined){
-        @if(balance.getOrElse(throw MissingFieldException("Balance")).availableCredit.getOrElse(throw MissingFieldException("Available Credit")) != 0){
+        @if(maybeUnallocatedCreditType(creditCharges, balance).contains(UnallocatedCreditFromOnePayment)) {
```

Comment:
actual call to maybeUnallocatedCreditType can be done of the controller level;
I was also expecting simple pattern matching instead of using if/else statement, as we have nesting statements etc, this would be a hard to support/read;

Reply:
Thanks it make sense. Hi I've just refactored code and pushed the fix. thanks